wolvic/browser/session_settings.cc

```diff
@@ -117,13 +84,36 @@ void SessionSettings::RenderViewHostChanged(content::RenderViewHost* old_host,
                                             content::RenderViewHost* new_host) {
   DCHECK_EQ(new_host, web_contents()->GetRenderViewHost());
 
-  //UpdateEverything();
+  UpdateEverything();
 }
 
-void SessionSettings::WebContentsDestroyed() {
-  // The destroyed WebContents instance is removed from the Observers
-  // lists, calling to the ResetWebContents private functtion, which
-  // assignes nullptr to the web_contents_ attribute.
+void SessionSettings::UpdateEverything() {
+  UpdateUserAgent();
+}
```

Comment:
Why do we need an additional method?

Reply:
Just a template for future settings that we may need to update when the rvh changes. Anyway, I just wanted to get early feedback on the general approach, whether we want / need to add this logic to the SessionSettings class.